### PR TITLE
Edit sigmoid_binary_cross_entropy and sigmoid_focal_loss to work correctly with non-array label arguments.

### PR DESCRIPTION
**Pull request by carlosgmartin**
Edit [sigmoid_binary_cross_entropy](https://optax.readthedocs.io/en/latest/api/losses.html#optax.losses.sigmoid_binary_cross_entropy) and [sigmoid_focal_loss](https://optax.readthedocs.io/en/latest/api/losses.html#optax.losses.sigmoid_focal_loss) to work correctly with non-array `label` arguments by replacing `x.astype(dt)` with `jnp.astype(x, dt)`.

### CODE REVIEW

**Review by vroulet — APPROVED** [review]
Good catch, thank you
